pkg/resource/plugin/provider.go

```diff
@@ -123,3 +123,8 @@ func DiffUnavailable(reason string) DiffUnavailableError {
 func (e DiffUnavailableError) Error() string {
 	return e.reason
 }
+
+type ReadResult struct {
```

Comment:
This could use a doc comment explaining why we return both inputs and outputs, since that's not obvious.

Reply:
Ah, good catch. I will add a doc comment.